nested-types.ipynb

```diff
@@ -0,0 +1,802 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Nested types\n",
+    "\n",
+    "* List, List View\n",
+    "* Struct\n",
+    "* Map\n",
+    "* Union\n",
+    "* Dictionary encoded,\n",
+    "* Run-end encoded\n",
```

Comment:
While strictly speaking run-end encoded is a nested type (in the sense of "has children"), for the tutorial I wouldn't consider it as nested here, but treat them separately.

Because while the encoded ones are nested in the physical layout, they can "logically" present a normal non-nested array (like run-end-encoded integer array, which is essentially a way to more efficiently store a plain integer array)